File: backend/AdSetLeadDesafio/DAL/DTOs/VeiculoFiltroDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DAL.DTOs
{
    public class VeiculoFiltroDTO
    {
        public string? Marca { get; set; }
        public string? Modelo { get; set; }
        public int? AnoMin { get; set; }
        public int? AnoMax { get; set; }
        public string? Placa { get; set; }
        public int? Km { get; set; }
        public string? Cor { get; set; }
        public int? Preco { get; set; }
        public int? Fotos { get; set; }
        public string? Opcionais { get; set; }
    }
}

File: frontend/src/app/models/veiculoFiltroDTO.ts
export class VeiculoFiltroDTO {
    public placa?: string | null = '';
    public marca?: string | null = '';
    public modelo?: string | null = '';
    public anoMin?: number | null = null;
    public anoMax?: number | null = null;
    public preco?: number | null = null;
    public fotos?: number | null = null;
    public opcionais?: string | null = '';
    public cor?: number | null = null;
}